src/diff/children.js

```diff
@@ -48,8 +47,8 @@ export function diffChildren(parentDom, newParentVNode, oldParentVNode, context,
 		}
 	}
 
-	for (i=0; i<newChildren.length; i++) {
-		childVNode = newChildren[i] = coerceToVNode(newChildren[i]);
+	i=0;
+	newParentVNode._children = toChildArray(newParentVNode._children, childVNode => {
```

Comment:
@andrewiggins any chance you tried checking the size with this inner function pulled out? I feel like the argument passing to account for locals would be a nightmare though.

Reply:
I did not. I think I talked myself out of that because, assuming `toChildArray` works as a callback, it'd significantly complicate this code. We'd have to capture all the args to call the callback with in an array, add the `childVNode` to that list, and then call `.apply` on the callback. So something like:

```js
toChildArray(vnode._children, diffChild, [...captured locals...]);

function diffChild(...capturedArgs..., childVNode) {
    // actual diff code
}

function toChildArray(children, callback, args) {
    if (...) {
        args.push(childVNode);
        flattened.push(callback.apply(null, args));
    }
    else {
         // similar code to above
    }
}
```

This felt like a lot of ceremony and I didn't think it'd save us bytes. If you think it might, I'd be happy to give it a shot.

For this PR I also didn't consider a non-recursive solution cuz in my small experience they've been bigger but perhaps that's worth an exploration in the future?